diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/StatelessClassesRule.kt

```diff
@@ -0,0 +1,99 @@
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.psi.KtClass
+
+/**
+ * This rule checks if class is stateless and if so changes it to object.
+ */
+class StatelessClassesRule(private val configRule: List<RulesConfig>) : Rule("stateless-class") {
+    private lateinit var emitWarn: EmitType
+    private var isFixMode: Boolean = false
+
+    override fun visit(
+            node: ASTNode,
+            autoCorrect: Boolean,
+            emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val interfaces: MutableList<ASTNode> = mutableListOf()
+
+        // Fixme: We should find interfaces in all project and then check them
+        if (node.elementType == FILE) {
+            val interfacesNodes = node
+                    .findAllNodesWithSpecificType(CLASS)
+                    .filter { it.hasChildOfType(INTERFACE_KEYWORD) }
+            interfacesNodes.forEach { interfaces.add(it) }
+            (node.findAllNodesWithSpecificType(CLASS) subtract interfacesNodes)
+                    .forEach{handleClass(it, interfaces)}
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleClass(node: ASTNode, interfaces: MutableList<ASTNode>) {
+        if (isClassExtendsValidInterface(node, interfaces) && isStatelessClass(node)) {
+            Warnings.OBJECT_IS_PREFERRED.warnAndFix(configRule, emitWarn, isFixMode,
+                    "class ${(node.psi as KtClass).name!!}", node.startOffset, node) {
+                val newObjectNode = CompositeElement(OBJECT_DECLARATION)
+                node.treeParent.addChild(newObjectNode, null)
+                node.removeChild(node.firstChildNode)
+                newObjectNode.addChild(LeafPsiElement(OBJECT_KEYWORD, "object"), null)
+                node.children().forEach {
+                    newObjectNode.addChild(it.copyElement(), null)
+                }
+                node.treeParent.removeChild(node)
+            }
+        }
+    }
+
+    private fun isStatelessClass(node: ASTNode): Boolean {
+        val properties = (node.psi as KtClass).getProperties()
+        val functions = node.findAllNodesWithSpecificType(FUN)
+        return properties.isNullOrEmpty()
+                && functions.isNotEmpty()
+                && !(node.psi as KtClass).hasExplicitPrimaryConstructor()
+    }
+
+    private fun isClassExtendsValidInterface(node: ASTNode, interfaces: MutableList<ASTNode>) : Boolean =
+            node.findChildByType(SUPER_TYPE_LIST)
+                ?.getAllChildrenWithType(SUPER_TYPE_ENTRY)
+                ?.isNotEmpty()
+                ?: false
+                    && isValidInterface(node, interfaces)
```

Comment:
```suggestion
                ?.isNotEmpty()
                ?.and(isValidInterface(node, interfaces))
                ?: false
```
This might be more readable, and the reader doesn't need to keep in mind operators priority.